pkg/server/service.go

```diff
@@ -27,11 +34,56 @@ import (
 
 type Server struct {
 	pb.UnimplementedRekorServer
+	storage *tessera.Storage
 }
 
-func (s *Server) CreateEntry(context.Context, *pb.CreateEntryRequest) (*pbs.TransparencyLogEntry, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method CreateEntry not implemented")
+func NewServer(storage *tessera.Storage) *Server {
+	return &Server{storage: storage}
 }
+
+func (s *Server) CreateEntry(ctx context.Context, req *pb.CreateEntryRequest) (*pbs.TransparencyLogEntry, error) {
+	var serialized []byte
+	var err error
+	switch req.GetSpec().(type) {
+	case *pb.CreateEntryRequest_HashedRekordRequest:
+		hr := req.GetHashedRekordRequest()
+		if err := hashedrekord.Validate(hr); err != nil {
+			slog.Warn("failed validating hashedrekord request", "error", err.Error())
+			return nil, status.Errorf(codes.InvalidArgument, "invalid hashedrekord request")
+		}
+		serialized, err = json.Marshal(hr)
```

Comment:
Do we need to use https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson#Marshal?

Reply:
I don't think so because `hr` is a regular struct not a `proto.Message`. @loosebazooka ?

Reply:
It looks like this does implement `proto.Message`, with the struct implementing [these methods](https://pkg.go.dev/google.golang.org/protobuf/runtime/protoiface#MessageV1). I think `json.Marshal` can work for some structs, but not if the struct contains any proto-specific messages like one-ofs.

Reply:
Oops you're right, switching it